### DISCUSSION TITLE
Check for permission in other team than currently logged in.

### DISCUSSION THREAD

**Discussion by Ximerox** [spatie/laravel-permission#2141] [Q&A] [answered]
Hello, basically everything works quite well. I use the teams function. I get the permissions and roles of the current team reliably. But I would like to check in a controller if the current user has a special permission in another team. E.g. user Tim is logged into team A and I would now like to check if he has a specific permission in teams B,C and D. Unfortunately I do not get this done. Even if I set the other team before with setPermissionsTeamId(), $user->HasPermissionTo() or $user->can() always returns only the permissions of the current team. What am I doing wrong?
Ximix

**Comment by erikn69** [comment, marked as answer]
Maybe this scopes could help you
https://github.com/spatie/laravel-permission/pull/1948/files
Also this trick could help
https://github.com/spatie/laravel-permission/pull/1845/files

**Reply by Ximerox** [reply]
Hello and thank you for the quick reply. The first link seems to be exactly what I am looking for. I will look into it.
Thanks

**Reply by Ximerox** [reply]
Hello,
have tested it extensively. Unfortunately without success. Probably it is beyond my PHP skills. I always get the error: "Class name must be a valid object or a string". I can't even determine which line in the code this error refers to....

**Reply by Ximerox** [reply]
> > **Ximerox** Unmarked an **Answer**
> 
> Do not expect more answers from me, good luck

Sorry, I didn't mean anything by that, I just wanted to add something. It seems to be already the solution, but I am probably not able to implement this.

**Reply by erikn69** [reply]
Make a new discussion posting the code, or a example on a minimal application on fresh installation with the exact exception
